src/app/[locale]/incident/add/components/questions/CheckboxInput.tsx

```diff
@@ -2,45 +2,91 @@ import { QuestionField } from '@/types/form'
 import { useTranslations } from 'next-intl'
 import { useFormStore } from '@/store/form_store'
 import { getValidators } from '@/lib/utils/form-validator'
-import React from 'react'
+import React, { useEffect, useMemo, useState } from 'react'
+import { evaluateConditions } from '@/lib/utils/check-visibility'
+import { useFormContext } from 'react-hook-form'
 import { Paragraph } from '@/components/index'
 
 interface CheckboxInputProps extends QuestionField {}
 
-export const CheckboxInput = ({
-  field,
-  register,
-  errors,
-}: CheckboxInputProps) => {
+export const CheckboxInput = ({ field }: CheckboxInputProps) => {
+  const [shouldRender, setShouldRender] = useState<boolean>(false)
+  const {
+    watch,
+    formState: { errors },
+    register,
+    setValue,
+  } = useFormContext()
   const t = useTranslations('general.errors')
-  const { formState } = useFormStore()
+  const { formState: formStoreState } = useFormStore()
 
   const errorMessage = errors[field.key]?.message as string
 
-  // Check if the user has already answered a specific question.
-  // Returns true if an answer exists where the id is same as the value key.
-  // This is used to determine if the 'defaultChecked' property of a checkbox input should be set.
-  const getDefaultValueCheckboxInput = (id: string, key: string) => {
-    const extraProperties = formState.extra_properties.filter(
+  const watchValues = watch()
+
+  // Memoize `evaluateConditions` result to prevent unnecessary updates
+  const shouldRenderResult = useMemo(
+    () => evaluateConditions(field.meta, watchValues),
+    [field.meta, watchValues]
+  )
+
+  // Handle visibility changes
```

Comment:
Het bepalen van de zichtbaarheid staat nu per component gerepliceerd in de code (5x). Zou het mogelijk zijn om deze logica één niveau hoger te tillen en in `IncidentQuestionsLocationForm` éénmaal te implementeren?

Reply:
Done :) Geïmplementeerd in ```RenderSingleField.tsx```